ktor-client/ktor-client-features/ktor-client-logging/jvm/src/io/ktor/client/features/logging/LoggerJvm.kt

```diff
@@ -10,3 +10,45 @@ actual val Logger.Companion.DEFAULT: Logger
             delegate.info(message)
         }
     }
+
+
+/***
+ * Workaround Logger to log long messages that would otherwise be truncated in Android.
+ */
+val Logger.Companion.ANDROID: Logger
+    get() = MaxLengthLogger()
+
+
+class MaxLengthLogger(val MAX_INDEX : Int = 4000) : Logger {
+    private val delegate = LoggerFactory.getLogger(HttpClient::class.java)!!
+    override fun log(message: String) {
+        logLong(message)
+    }
+
+    fun logLong(message: String) {
+
+        val MIN_INDEX = 3000
+
+        // String to be logged is longer than the max...
+        if (message.length > MAX_INDEX) {
+            var theSubstring = message.substring(0, MAX_INDEX)
+            var theIndex = MAX_INDEX
+
+            // Try to find a substring break at a line end.
+            theIndex = theSubstring.lastIndexOf('\n')
+            if (theIndex >= MIN_INDEX) {
+                theSubstring = theSubstring.substring(0, theIndex)
+            } else {
+                theIndex = MAX_INDEX
+            }
+
+            // Log the substring.
+            delegate.info(theSubstring)
+
+            // Recursively log the remainder.
+            logLong(message.substring(theIndex))
```

Comment:
Shouldn't we discard leading newline character in the first case when it is found?

Reply:
yeah, didn't catch that